### PR TITLE
libstrophe: 0.10.0 -> 0.10.1

### PR DESCRIPTION
**Pull request by devhell**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rmcgibbo** [comment]
Result of `nixpkgs-review pr 107644` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libstrophe</li>
  </ul>
</details>

File: pkgs/development/libraries/libstrophe/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 
   src = fetchFromGitHub {
     owner = "strophe";
     repo = pname;
     rev = version;
-    sha256 = "1hizw695fw0cy88h1dpl9pvniapml2zw9yvxck8xvxbqfz54jwja";
+    sha256 = "11d341avsfr0z4lq15cy5dkmff6qpy91wkgzdpfdy31l27pa1g79";
   };
 
   nativeBuildInputs = [ automake autoconf pkgconfig libtool check ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libstrophe/default.nix:14]:
```suggestion
  nativeBuildInputs = [ automake autoconf pkg-config libtool check ];
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107644` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>libstrophe</li>
  </ul>
</details>

```
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_stdio.h:57:39: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
                                      ^
tests/test_rand.c:37:5: note: to match this '('
./src/snprintf.h:23:23: note: expanded from macro 'xmpp_snprintf'
#define xmpp_snprintf snprintf
                      ^
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_stdio.h:57:28: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
                           ^
tests/test_rand.c:37:5: error: conflicting types for '__builtin___snprintf_chk'
int xmpp_snprintf(char *str, size_t count, const char *fmt, ...)
    ^
./src/snprintf.h:23:23: note: expanded from macro 'xmpp_snprintf'
#define xmpp_snprintf snprintf
                      ^
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_stdio.h:57:3: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
  ^
tests/test_rand.c:37:5: note: '__builtin___snprintf_chk' is a builtin with type 'int (char *, unsigned long, int, unsigned long, const char *, ...)'
./src/snprintf.h:23:23: note: expanded from macro 'xmpp_snprintf'
#define xmpp_snprintf snprintf
                      ^
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_stdio.h:57:3: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
  ^
tests/test_rand.c:37:5: error: definition of builtin function '__builtin___snprintf_chk'
int xmpp_snprintf(char *str, size_t count, const char *fmt, ...)
    ^
./src/snprintf.h:23:23: note: expanded from macro 'xmpp_snprintf'
#define xmpp_snprintf snprintf
                      ^
/nix/store/r913a0b23i21r8y4x8kgx024gms9lvx9-Libsystem-osx-10.12.6/include/secure/_stdio.h:57:3: note: expanded from macro 'snprintf'
  __builtin___snprintf_chk (str, len, 0, __darwin_obsz(str), __VA_ARGS__)
  ^
tests/test_rand.c:41:11: error: use of undeclared identifier 'fmt'
    (void)fmt;
          ^
5 errors generated.
make[1]: *** [Makefile:1832: tests/test_rand-test_rand.o] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-libstrophe-0.10.1.drv-0/source'
make: *** [Makefile:2654: check-am] Error 2
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107644` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libstrophe</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107644` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libstrophe</li>
  </ul>
</details>

File: pkgs/development/libraries/libstrophe/default.nix [hunk 2]
```diff
@@ -1,35 +1,33 @@
...
     repo = pname;
     rev = version;
-    sha256 = "1hizw695fw0cy88h1dpl9pvniapml2zw9yvxck8xvxbqfz54jwja";
+    sha256 = "11d341avsfr0z4lq15cy5dkmff6qpy91wkgzdpfdy31l27pa1g79";
   };
 
-  nativeBuildInputs = [ automake autoconf pkgconfig libtool check ];
-  buildInputs = [ openssl expat ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
+  buildInputs = [ openssl expat libtool check ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libstrophe/default.nix:15]: I am not sure if lobtool and check belong into nativeBuildInputs. Are they required at runtime?

Comment (devhell) [reply pkgs/development/libraries/libstrophe/default.nix:23]: No, you are correct. `libmesode` is a fork of `libstrophe`, but I've not had the chance to sync them up until now. However, as with `libmesode` I cannot investigate the darwin issues.

**Review by devhell — COMMENTED** [review]
Thanks @SuperSandro2000, hope this suffices.

File: pkgs/development/libraries/libstrophe/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libstrophe/default.nix:23]: No problem. Can you please mark it as broken on darwin? So I won't nag you about this on the next update.

Comment (devhell) [reply pkgs/development/libraries/libstrophe/default.nix:23]: > No problem. Can you please mark it as broken on darwin? So I won't nag you about this on the next update.

All done :)

**Comment by devhell** [comment]
@SuperSandro2000 does this need more work or can it be merged?

**Comment by devhell** [comment]
Bump?

**Comment by devhell** [comment]
@SuperSandro2000 could you please let me know if this can be merged or if there's anything left to do? Thank you :)

**Comment by SuperSandro2000** [comment]
@devhell please solve the merge conflict.

**Comment by devhell** [comment]
@SuperSandro2000 Let me know if this is okay now, please. Thank you

File: pkgs/development/libraries/libstrophe/default.nix [hunk 3]
```diff
@@ -37,8 +33,10 @@ stdenv.mkDerivation rec {
...
     homepage = "https://strophe.im/libstrophe/";
-    license = with licenses; [ gpl3 mit ];
+
+    license = with licenses; [ gpl3Only mit ];
     platforms = platforms.unix;
+    broken = stdenv.isDarwin;
     maintainers = with maintainers; [ devhell flosse ];
   };
-}
+}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libstrophe/default.nix:42]: Add the final new line back.

File: pkgs/development/libraries/libstrophe/default.nix [hunk 4]
```diff
@@ -37,8 +33,10 @@ stdenv.mkDerivation rec {
       runs well on both Linux, Unix, and Windows based platforms.
     '';
     homepage = "https://strophe.im/libstrophe/";
-    license = with licenses; [ gpl3 mit ];
+
+    license = with licenses; [ gpl3Only mit ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libstrophe/default.nix:37]:
```suggestion
    license = with licenses; [ gpl3Only mit ];
```

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 Let me know if this is okay now, please. Thank you

Merging master into a PR is big no go and I can't merge something that does it.

File: pkgs/development/libraries/libstrophe/default.nix (hunk 3 above)
Comment (devhell) [reply pkgs/development/libraries/libstrophe/default.nix]: Done.

**Comment by devhell** [comment]
> > @SuperSandro2000 Let me know if this is okay now, please. Thank you
> 
> Merging master into a PR is big no go and I can't merge something that does it. Please use rebase instead.

Dammit. That was unintentional, sorry.

File: pkgs/development/libraries/libstrophe/default.nix [hunk 5]
```diff
@@ -37,8 +34,10 @@ stdenv.mkDerivation rec {
...
     '';
     homepage = "https://strophe.im/libstrophe/";
-    license = with licenses; [ gpl3 mit ];
+    license = with licenses; [ gpl3Only mit ];
     platforms = platforms.unix;
+    broken = stdenv.isDarwin;
     maintainers = with maintainers; [ devhell flosse ];
   };
 }
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libstrophe/default.nix:43]:
```suggestion
}
```

**Pull request merged**